contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -382,10 +384,23 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
         factory = ValidatorShareFactory(newFactory);
     }
 
-    function updateValidatorState(uint256 validatorId, int256 amount) public {
-        require(validators[validatorId].contractAddress == msg.sender, "Invalid contract address");
+    function updateValidatorState(uint256 validatorId, int256 amount) public onlyDelegation(validatorId) {
         uint256 _currentEpoch = currentEpoch;
         validatorState[_currentEpoch].amount = (validatorState[_currentEpoch].amount + amount);
+        
+        if (amount >= 0) {
```

Comment:
why this is required?

Reply:
in case of jail/unjail?

Reply:
Yes, will do >, < checks to avoid unnecessary call

Reply:
why are you updating the delegation amount in this function?

Reply:
Because before it was used everywhere as additional call next to `activeAmount` modifications. Now there is not activeAmount but `delegatedAmount` in the Validator struct. So i just moved that logic inside `updateValidatorState`